server/api/views/user.py

```diff
@@ -1,11 +1,50 @@
-from django.contrib.auth.models import User, Group
-from rest_framework import viewsets
-from api.serializers import UserSerializer, GroupSerializer
+from django.http import HttpResponseRedirect
+from django.contrib.auth.models import User
+from rest_framework import permissions, status
+from rest_framework.decorators import api_view
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from api.serializers import UserSerializer, GroupSerializer, UserSerializerWithToken
 
+# https://medium.com/@dakota.lillie/django-react-jwt-authentication-5015ee00ef9a
+@api_view(['GET'])
+def current_user(request):
+    """
+    Determine the current user by their token, and return their data
+    """
 
-class UserViewSet(viewsets.ModelViewSet):
+    serializer = UserSerializer(request.user)
+    return Response(serializer.data)
+
+# https://medium.com/@dakota.lillie/django-react-jwt-authentication-5015ee00ef9a
+class UserList(APIView):
     """
-    API endpoint that allows users to be viewed or edited.
+    Create a new user. It's called 'UserList' because normally we'd have a get
+    method here too, for retrieving a list of all User objects.
     """
-    queryset = User.objects.all().order_by('-date_joined')
-    serializer_class = UserSerializer
+
+    permission_classes = (permissions.AllowAny,)
+
+    def post(self, request, format=None):
+        serializer = UserSerializerWithToken(data=request.data)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_201_CREATED)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+
+"""
```

Comment:
You can go ahead and remove code from previous git revisions instead of commenting it out. If it's needed in the future it will still be in git history